Refactors Redux logic and improves code quality

Da1ryman opened the pull request.

Da1ryman committed:
Refactors Redux logic and improves code quality

Da1ryman committed:
Refactors store structure for better organization

Moves news and comment related actions and reducers into separate slices.
This change improves code maintainability and scalability by grouping related logic together.
Removes the old action and slice files.

XeniaVi commented on src/api/HackerNewsAPI.ts:
```diff
@@ -1,31 +1,29 @@
 
-export const getNewStoriesId = async (): Promise<number[]> => {
-  const response = await axios.get<number[]>(
+export const getNewStoriesId = async () => {
```
теперь возвразается Promise<any> и смысл ts вообще теряется

XeniaVi commented on src/components/news/NewsDetail.tsx:
```diff
@@ -26,30 +23,28 @@ export const NewsDetail = () => {
+        </Card.Text>
+
+        <div className='d-flex justify-content-between '>
+          <Button as={Link as any} to={newsDetail.url}>
```
использование as  итак не очень хорошая практика, но с any  так вообще

XeniaVi commented on src/components/news/NewsDetail.tsx:
```diff
@@ -26,30 +23,28 @@ export const NewsDetail = () => {
+            Посмотреть новость
+          </Button>
+
+          <Button as={Link as any} to='/' onClick={removeComments}>
```
и ссылка и кнопка, странно это выглядит, тогда просто на клик делай навигацию после очищения стора

XeniaVi commented on src/index.tsx:
```diff
@@ -10,8 +10,9 @@ import { NewsPage } from './screen/news/NewsPage';
+if (!rootElement) {
+  throw new Error('Root element not found');
+}
 const root = ReactDOM.createRoot(rootElement);
```
отступ между

Da1ryman committed:
Improves type safety and navigation flow

Enhances type safety by explicitly defining types for API responses using generics in axios requests.

Refactors navigation in the NewsDetail component for improved user experience.
It replaces the react-router-dom Link component with the useNavigate hook to allow dispatching redux actions before navigating.

Updates tsconfig to disallow implicit any types.

XeniaVi commented on src/components/news/NewsDetail.tsx:
```diff
@@ -1,15 +1,21 @@
 
   const { newsDetail, loadingDetail } = useAppSelector((state) => state.news);
 
+  const navigation = (url: string) => {
```
handleNavigation

Da1ryman committed:
Renames navigation function for clarity.

Renames the `navigation` function to `handleNavigation` to
better reflect its purpose and improve code readability.
